artemis-protocols/artemis-amqp-protocol/src/main/java/org/apache/activemq/artemis/protocol/amqp/proton/handler/EventHandler.java

```diff
@@ -91,4 +91,7 @@ default boolean flowControl(ReadyListener readyListener) {
       return true;
    }
 
+   default String getRemoteAddress() {
```

Comment:
You don't need this. you're using AMQPConnection directly, no need to change to change EventHandler.

Can you remove it?

Reply:
actually, you do use it...

But you are calling this on every proton event. isn't there a better place to call that?

Reply:
It is on every event so it is available to the Audit logger